go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -181,9 +221,13 @@ func getFlightClient(ctx context.Context, loc string, d *database) (*flightsql.C
 	if err != nil {
 		return nil, adbc.Error{Msg: fmt.Sprintf("Invalid URI '%s': %s", loc, err), Code: adbc.StatusInvalidArgument}
 	}
+	creds := d.creds
+	if uri.Scheme == "grpc" || uri.Scheme == "grpc+tcp" {
+		creds = insecure.NewCredentials()
+	}
 
 	cl, err := flightsql.NewClient(uri.Host, nil, []flight.ClientMiddleware{
-		flight.CreateClientMiddleware(authMiddle)}, grpc.WithTransportCredentials(d.creds))
+		flight.CreateClientMiddleware(authMiddle)}, grpc.WithTransportCredentials(creds))
```

Comment:
why bother with the separate `creds` var instead of just using `d.creds` ?

Reply:
I wanted to have it so that you can set TLS options but still connect to an unencrypted backend, in case the backend in turn returns locations using TLS

Reply:
ah gotcha, that makes sense.